contracts/external/dao-voting-incentives/src/tests.rs

```diff
@@ -0,0 +1,470 @@
+    context
+        .app
+        .execute_contract(
+            Addr::unchecked(ADDR1),
+            context.proposal_single_addr.clone(),
+            &dao_proposal_single::msg::ExecuteMsg::Propose(SingleChoiceProposeMsg {
+                title: "Test proposal".to_string(),
+                description: "Testing".to_string(),
+                msgs: vec![],
+                proposer: None,
+            }),
+            &[],
+        )
+        .unwrap();
+
+    // Trigger a vote hook
+    let results = vote_yes_on_proposal(&mut context, 2u64);
+    for result in results {
+        assert!(result.is_ok());
+    }
+
+    // Assert that the vote hook has incremented vote counts
+    let votes: Uint128 = context
+        .app
+        .wrap()
+        .query_wasm_smart(
+            dao_voting_incentives_addr.clone(),
+            &QueryMsg::Votes {
+                address: ADDR1.to_string(),
+            },
+        )
+        .unwrap();
+    assert_eq!(votes, Uint128::one());
+    let votes: Uint128 = context
+        .app
+        .wrap()
+        .query_wasm_smart(
+            dao_voting_incentives_addr.clone(),
+            &QueryMsg::Votes {
+                address: ADDR2.to_string(),
+            },
+        )
+        .unwrap();
+    assert_eq!(votes, Uint128::one());
+    let votes: Uint128 = context
+        .app
+        .wrap()
+        .query_wasm_smart(
+            dao_voting_incentives_cw20_addr.clone(),
+            &QueryMsg::Votes {
+                address: ADDR1.to_string(),
+            },
+        )
+        .unwrap();
+    assert_eq!(votes, Uint128::one());
+    let votes: Uint128 = context
+        .app
+        .wrap()
+        .query_wasm_smart(
+            dao_voting_incentives_cw20_addr.clone(),
+            &QueryMsg::Votes {
+                address: ADDR2.to_string(),
+            },
+        )
+        .unwrap();
+    assert_eq!(votes, Uint128::one());
+    let config: Config = context
+        .app
+        .wrap()
+        .query_wasm_smart(dao_voting_incentives_addr.clone(), &QueryMsg::Config {})
+        .unwrap();
+    assert_eq!(config.total_votes, Uint128::new(2));
+
+    // Blocks have passed the voting incentives' expirations
+    context.app.update_block(|x| x.height += 100);
+
+    // Creating another proposal and voting should still succeed but unregister these vote hooks
+    context
+        .app
+        .execute_contract(
+            Addr::unchecked(ADDR1),
+            context.proposal_single_addr.clone(),
+            &dao_proposal_single::msg::ExecuteMsg::Propose(SingleChoiceProposeMsg {
+                title: "Test proposal".to_string(),
+                description: "Testing".to_string(),
+                msgs: vec![],
+                proposer: None,
+            }),
+            &[],
+        )
+        .unwrap();
+    let results = vote_yes_on_proposal(&mut context, 3u64);
+    for (i, result) in results.iter().enumerate() {
+        assert!(result.is_ok());
```

Comment:
This actually panics, because the hooks simultaneously return an error of expiration.
When the error is processed in the proposal module, it tries to remove by index on a list that has changed already.

```
thread 'tests::test_hooks' panicked at packages/cw-hooks/src/lib.rs:60:26:
removal index (is 1) should be < len (is 1)
```

VoteHook of index 0 is successfully removed at index 0, but VoteHook of index 1 tries to be removed at index 1 of a list of length 1.

@JakeHartnell @NoahSaso Thoughts?

Reply:
I added a new method in cw-hooks for `remove_hook_by_index_from_reply` that takes a new parameter `removed_hook_indexes: Item<Vec<u64>>`. 

We can adjust the given index to the actual index from this new tracked state.